tests/db_functions/comparison/test_cast.py

```diff
@@ -88,6 +88,28 @@ def test_cast_from_db_date_to_datetime(self):
         dtm = DTModel.objects.annotate(start_as_datetime=Cast('start_date', models.DateTimeField())).first()
         self.assertEqual(dtm.start_as_datetime, datetime.datetime(2018, 9, 28, 0, 0, 0, 0))
 
+    def test_cast_to_db_duration(self):
+        duration1_value = datetime.timedelta(
+            days=50,
+            seconds=27,
+            microseconds=10,
+            milliseconds=29000,
+            minutes=5,
+            hours=8,
+            weeks=2,
+        )
+        duration2_value = datetime.timedelta(
+            days=1,
+            seconds=2,
+            microseconds=3,
+        )
+        DTModel.objects.create(duration=duration1_value)
+        DTModel.objects.create(duration=duration2_value)
+        dtm = DTModel.objects.aggregate(
+            sum_of_durations=Cast(models.Sum('duration'), models.DurationField())
+        )
+        self.assertEqual(dtm['sum_of_durations'], datetime.timedelta(days=65, seconds=29158, microseconds=13))
```

Comment:
Thinking a bit more about we might want to add a test for a negative duration as well to cover the `unsigned` part

```diff
diff --git a/tests/db_functions/comparison/test_cast.py b/tests/db_functions/comparison/test_cast.py
index 40d61bb868..a5bc18c4e1 100644
--- a/tests/db_functions/comparison/test_cast.py
+++ b/tests/db_functions/comparison/test_cast.py
@@ -89,7 +89,7 @@ class CastTests(TestCase):
         self.assertEqual(dtm.start_as_datetime, datetime.datetime(2018, 9, 28, 0, 0, 0, 0))

     def test_cast_to_db_duration(self):
-        duration1_value = datetime.timedelta(
+        duration = datetime.timedelta(
             days=50,
             seconds=27,
             microseconds=10,
@@ -98,17 +98,13 @@ class CastTests(TestCase):
             hours=8,
             weeks=2,
         )
-        duration2_value = datetime.timedelta(
-            days=1,
-            seconds=2,
-            microseconds=3,
-        )
-        DTModel.objects.create(duration=duration1_value)
-        DTModel.objects.create(duration=duration2_value)
-        dtm = DTModel.objects.aggregate(
-            sum_of_durations=Cast(models.Sum('duration'), models.DurationField())
-        )
-        self.assertEqual(dtm['sum_of_durations'], datetime.timedelta(days=65, seconds=29158, microseconds=13))
+        DTModel.objects.create(duration=duration)
+        dtm = DTModel.objects.annotate(
+            duration_cast=Cast(duration, models.DurationField()),
+            neg_duration_cast=Cast(-duration, models.DurationField()),
+        ).first()
+        self.assertEqual(dtm.duration_cast, duration)
+        self.assertEqual(dtm.neg_duration_cast, -duration)

     def test_cast_from_db_datetime_to_date_group_by(self):
         author = Author.objects.create(name='John Smith', age=45)
```

Reply:
Thank you for suggestions!

> 
> In order to make the regression test as succinct as possible I'd suggest you use the smallest reproducible scenario instead of creating two objects and using aggregation. 
> 

I simplified test. It is also failing without patch. The original test was showing "real" scenario, where cast to duration is needed, but I agree that simpler test is better.

> 
> Thinking a bit more about we might want to add a test for a negative duration as well to cover the `unsigned` part
> 

I did that as well, however using `-duration` without quotes just supplies integer to the sql query and it is not really testing db_to_duration cast. I tried to use:

    neg_duration_cast=Cast(0-models.F('duration'), models.DurationField()),

but it is failing on both sqlite and mysql from various reasons. Mathematics operations are clearly not properly defined for timedelta/DurationField and fixing it will require much more code. I suggest to stick with this simple test.